### PR TITLE
Add more support for entitlements (`ReferenceType.Authorization`) to CCF codec

### LINKED ISSUE #2379: Update CCF codec to support recently added types/values (e.g. AttachmentType)
### Issue to be solved

New Cadence types and/or values were recently added, such as `AttachmentType`, `AnyStructAttachmentType`, `AnyResourceAttachementType`.

They need to be supported in CCF Specs and CCF Codec.

### Suggested Solution

NOTE: The PR for this depends on PR #2364 (CCF Codec) being approved and merged first.

First, update [CCF Specs](https://github.com/fxamacker/ccf_draft) to support new types (e.g.`AttachmentType`, `AnyStructAttachmentType`, `AnyResourceAttachementType`) and bump specs (e.g. from RC1->RC2).

Then update [CCF Codec](https://github.com/onflow/cadence/pull/2364) to match updated CCF Specs.

We're leaving room for possibility of [CCF fuzz testing](https://github.com/onflow/cadence/issues/2377) (and maybe integration tests) to uncover issues that require changes to CCF Specs and/or CCF Codec.

### PR DESCRIPTION
**Pull request by fxamacker**
Closes #2379

Support for `ReferenceType.Authorization` is needed for CCF codec to support entitlements.

Add support in CCF codec for:

- encode/decode `Unauthorized`, `EntitlementSetAuthorization`, `EntitlementMapAuthorization` in `ReferenceType`
- encode/decode `Unauthorized`, `EntitlementSetAuthorization`, `EntitlementMapAuthorization` in `TypeValue{ReferenceType}`

This PR also adds 
- new `EncOptions.SortEntitlementTypes` to sort entitlements
- new `DecOptions.EnforceSortEntitlementTypes` to enforce sorted entitlements

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by dsainati1 — APPROVED** [review]
Thanks for adding this!

File: encoding/ccf/decode_type.go [hunk 1]
```diff
@@ -347,9 +347,218 @@ func (d *Decoder) decodeCapabilityType(
...
+		return nil, err
+	}
+
+	var kind cadence.EntitlementSetKind
+
+	switch rawKind {
+	case uint64(cadence.Conjunction):
+		kind = cadence.Conjunction
+	case uint64(cadence.Disjunction):
+		kind = cadence.Disjunction
```
Comment (turbolent) [inline encoding/ccf/decode_type.go:480]: We might want to define an enum in CCF itself, so accidental re-ordering in the `cadence` package does not lead to change of semantics.

**Review by turbolent — APPROVED** [review]
Great work!

File: encoding/ccf/decode_type.go (hunk 1 above)
Comment (fxamacker) [reply encoding/ccf/decode_type.go]: Good idea!  Done in commit [e54700f](https://github.com/onflow/cadence/pull/3139/commits/e54700f90755c5402962d601e1417a2999ccfe60).

**Pull request merged**